rust/types/src/metadata.rs

```diff
@@ -97,42 +102,50 @@ MetadataValue
 ===========================================
 */
 
-#[derive(Clone, Debug, PartialEq)]
+#[derive(Clone, Debug, PartialEq, PartialOrd)]
 pub enum MetadataValue {
-    Int(i32),
+    Bool(bool),
+    Int(i64),
     Float(f64),
     Str(String),
-    Bool(bool),
 }
 
-impl TryFrom<&MetadataValue> for i32 {
+impl Eq for MetadataValue {}
+
+impl Ord for MetadataValue {
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.partial_cmp(other).unwrap_or(Ordering::Equal)
```

Comment:
nit: add a comment about why `partial_cmp` here v/s `cmp` (I guess it is to handle floats?)

Reply:
`partial_cmp` was derived for `MetadataValue` with `#[derive(PartialOrd)]` above, but `cmp` cannot be derived because of the `f64`. I hope to use `MetadataValue` as a key in `BTreeMap`, which requires `Ord`. I believe we are not allowing `f64::NaN`s anyway so a `partial_cmp` should be equivalent to `cmp` in our case.